internal/pkg/handler/create.go

```diff
@@ -0,0 +1,139 @@
+package handler
+
+import (
+	"encoding/json"
+	"errors"
+	logger "github.com/sirupsen/logrus"
+	"github.com/stakater/ProxyInjector/internal/pkg/callbacks"
+	"github.com/stakater/ProxyInjector/internal/pkg/constants"
+	"github.com/stakater/ProxyInjector/pkg/kube"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/util/retry"
+)
+
+// ResourceCreatedHandler contains new objects
+type ResourceCreatedHandler struct {
+	Resource interface{} `json:"resource"`
+}
+
+type Container struct {
+	Name  string   `json:"name"`
+	Image string   `json:"image"`
+	Args  []string `json:"args"`
+}
+
+type Spec2 struct {
+	Containers []Container `json:"containers"`
+}
+
+type Template struct {
+	Spec Spec2 `json:"spec"`
+}
+
+type Spec1 struct {
+	Tmpl Template `json:"template"`
+}
+
+type patch struct {
+	Spec Spec1 `json:"spec"`
+}
+
+// Handle processes the newly created resource
+func (r ResourceCreatedHandler) Handle(conf []string) error {
+	if r.Resource == nil {
+		logger.Errorf("Resource creation handler received nil resource")
+	} else {
+		name := callbacks.GetDeploymentName(r.Resource)
+		namespace := callbacks.GetDeploymentNamespace(r.Resource)
```

Comment:
You're assuming that the resource can only be a deployment. It should cater all resources like daemonsets and statefulsets.